transcom/mymove
# INT -  B 21655 alternative AK/HI duty location names

## [B-21655](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1029436)

## Summary

This PR addresses an issue related to searching AK and HI duty locations using common alternative names. It corrects initial migration introduced in [B-21481](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1016235&RoomContext=TeamRoom%3A855664&concept=TeamRoom). During migration some of the alternative names were erroneously migrated as official locations into the duty_locations table. This PR corrects it by deleting the records from the duty_locations table and moving it to the duty_location_name table. Each record in the duty_location_names will then be mapped to the official duty location.

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Edit .envrc and set FEATURE_FLAG_ENABLE_ALASKA=true

1. Login as customer and setup a move. Search for duty location. Use attached files for reference. Everything highlighted in column A represents a record that is to be in the duty_location_names table. Each new record in duty_location_names table will then be mapped to the official duty locations via foreign key. This mapping is represented in column B next to the official duty location in column A.
[AK-duty-locations-1730773768958 -Reviewed.xlsx](https://github.com/user-attachments/files/17671501/AK-duty-locations-1730773768958.-Reviewed.xlsx)
[HI-duty-locations-1730773801336 Reviewed.xlsx](https://github.com/user-attachments/files/17671502/HI-duty-locations-1730773801336.Reviewed.xlsx)
2. Search using any of the highlighted texts in spreadsheet. Verify returned results contain the official duty location.
For example:
Searching for "Stevens Vlg" should return a list containing "Stevens Village, AK 99774".
Searching for "JBPHH" should return a list containing "JB Pearl Harbor Hickam, HI 96853 " and "JB Pearl Harbor Hickam, HI 96860".


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:
